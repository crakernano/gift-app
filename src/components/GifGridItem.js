import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = (img)=>{
    return(
        <div className="card animate_fadeIn">
            <img src={img.url} alt={img.title} />
            <p>{img.title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}