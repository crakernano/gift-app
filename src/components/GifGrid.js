import React from 'react'
import { useFeatchGifs } from '../hooks/useFeatchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

//    const [images, setImages] = useState([]);


    const {data, loading} = useFeatchGifs( category );

    return (
        <div className="card-grid">
            <h3>{category}</h3>
            {loading ? 'Cargando...' : ''}
                {
                data.map(img => (
                    <GifGridItem 
                        key={img.id}
                        { ...img }
                    />
                ))
                }
        </div>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}