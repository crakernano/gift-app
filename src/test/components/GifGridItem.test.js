import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem'


describe('Pruebas de <GifGridItem />', () => {

    const title = 'un titulo'
    const url = 'https://localhost/prueba.html'
    const wrapper = shallow( <GifGridItem title={title} url={url}/>)

    test('Debe de mostrar el componente correctamente', ()=>{

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un parrafo con el texto', () =>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( title );
    })

    test('debe mostrar la imagen con la URL de la imagen', ()=>{
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );
    })

    test('Debe tener una clase para la animaciÃ³n', ()=>{
        const div = wrapper.find('div');
        const classname = div.prop('className');
        expect(classname.includes('animate_fadeIn')).toBe(true);

    })

})