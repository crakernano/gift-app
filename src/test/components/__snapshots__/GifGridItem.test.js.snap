// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate_fadeIn"
>
  <img
    alt="un titulo"
    src="https://localhost/prueba.html"
  />
  <p>
    un titulo
  </p>
</div>
`;
